puddlestuff/puddleobjects.py

```diff
@@ -654,20 +654,26 @@ def getfiles(files, subfolders=False):
             if not isdir(f):
                 yield f
             else:
-                dirname, subs, fnames = next(os.walk(f))
-                for fname in fnames:
-                    yield join(dirname, fname)
+                try:
+                    dirname, subs, fnames = next(os.walk(f))
+                    for fname in fnames:
+                        yield join(dirname, fname)
+                except StopIteration:
+                    pass
     else:
         for f in files:
             if not isdir(f):
                 yield f
             else:
-                for dirname, subs, fnames in os.walk(f):
```

Comment:
following my previous comment, this change hunk should likely be reverted

Reply:
Agreed, reverted.